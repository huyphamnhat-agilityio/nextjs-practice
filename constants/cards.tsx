// Components
import { HeroCardProps } from "@/components/common/HeroCard";
import {
  BlackBoardIcon,
  MortarboardIcon,
  TelescopeIcon,
} from "@/components/common/Icons";

export const HERO_CARD_LIST: Array<HeroCardProps> = [
  {
    icon: <BlackBoardIcon />,
    iconBackgroundColor: "primary",
    title: "Expert instruction",
    description:
      "The gradual accumulation of information about atomic and small-scale behaviour...",
  },
  {
    icon: <TelescopeIcon />,
    iconBackgroundColor: "secondary",
    title: "Training Courses",
    description:
      "The gradual accumulation of information about atomic and small-scale behaviour...",
  },
  {
    icon: <MortarboardIcon />,
    iconBackgroundColor: "default",
    title: "Lifetime access",
    description:
      "The gradual accumulation of information about atomic and small-scale behaviour...",
  },
];
