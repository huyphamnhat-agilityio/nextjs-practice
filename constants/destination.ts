export const DESTINATION = {
  HOME: "/",
  PRODUCT: "/products",
} as const;
