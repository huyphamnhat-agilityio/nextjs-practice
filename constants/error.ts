export const ERROR_MESSAGES = {
  400: "The request could not be understood by the server due to malformed syntax. Please check your input and try again.",
  401: "You are not authorized to access this resource. Please log in with valid credentials and try again.",
  403: "You do not have permission to access this resource. Please contact the administrator if you believe this is an error.",
  404: "The requested resource could not be found on this server. Please check the URL and try again.",
  405: "The HTTP method used is not allowed for this resource. Please refer to the API documentation for the correct method.",
  406: "The requested resource cannot generate content acceptable according to the Accept headers sent in the request. Please modify your request and try again.",
} as const;
