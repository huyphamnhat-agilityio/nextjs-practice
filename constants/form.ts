export const FORM_STATUS = {
  SUCCESS: "success",
  ERROR: "error",
} as const;
