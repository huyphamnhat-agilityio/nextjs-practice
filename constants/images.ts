import { Avatar } from "@/types";

export const IMAGES: Record<string, Avatar> = {
  HERO_BACKGROUND_THUMBNAIL: {
    src: "/hero-background-children.svg",
    alt: "An image of a child holding notebooks",
  },
  GET_QUALITY_SECTION_THUMBNAIL: {
    src: "/girl-holding-pencil-image.svg",
    alt: "An image of a girl holding a pencil",
  },
} as const;

export const MAX_IMAGE_FILE_SIZE = 32000000;

export const ACCEPTED_IMAGE_TYPES = [
  "image/jpeg",
  "image/jpg",
  "image/png",
  "image/webp",
];
