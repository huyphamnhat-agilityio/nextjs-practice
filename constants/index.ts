export * from "./nav-list";
export * from "./cards";
export * from "./resources";
export * from "./pagination";
export * from "./images";
export * from "./message";
export * from "./destination";
export * from "./tags";
export * from "./toast";
