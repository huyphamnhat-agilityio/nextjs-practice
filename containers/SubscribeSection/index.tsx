import { Input } from "@nextui-org/react";

// Components
import { Button } from "@/components";

const SubscribeSection = () => {
  return (
    <section>
      <div className="max-w-lg md:max-w-4xl lg:max-w-5xl my-0 mx-auto py-40 flex flex-col gap-20">
        <div className="flex flex-col gap-2.5 text-center">
          <h6 className="text-primary text-sm/6 font-bold">Newsletter</h6>
          <h3 className="text-foreground text-2xl font-bold">
            Most Popular Courses
          </h3>
          <p className="text-foreground-100 text-sm">
            Problems trying to resolve the conflict between
            <br />
            the two major realms of Classical physics: Newtonian mechanics
          </p>
        </div>
        <form className="flex flex-col gap-2 items-center">
          <Input
            type="email"
            placeholder="Your Email"
            classNames={{
              inputWrapper:
                "max-w-[80%] sm:max-w-172 self-center  bg-foreground-200 py-3.75 px-4 !rounded-1.25 border border-foreground-300 h-auto relative",
              input: "text-foreground-100 text-sm/7 h-auto",
            }}
            endContent={
              <Button
                size="md"
                className="hidden sm:block px-5.625 text-white text-sm/7 absolute right-0 rounded-l-none rounded-r-1.25"
              >
                Subscribe
              </Button>
            }
          />

          <Button
            size="md"
            className="sm:hidden px-5.625 text-white text-sm/7 w-full max-w-[80%]"
          >
            Subscribe
          </Button>
        </form>
      </div>
    </section>
  );
};

export default SubscribeSection;
