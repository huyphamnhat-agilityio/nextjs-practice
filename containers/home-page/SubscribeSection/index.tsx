"use client";
import { useFormState, useFormStatus } from "react-dom";
import { useEffect } from "react";
import toast from "react-hot-toast";
import { Control, Controller, FieldErrors, useForm } from "react-hook-form";
import { zodResolver } from "@hookform/resolvers/zod";

// Components
import { Button, Input } from "@/components";

// Services
import { subscribe } from "@/lib";

// Types
import { FormState, Subscription, SubscriptionForm } from "@/types";

// Constants
import { FORM_STATUS } from "@/constants";

// Schemas
import { SubscribeSchema } from "@/schemas";

export type SubscribeFormProps = {
  state: FormState<Subscription>;
  control: Control<SubscriptionForm, any>;
  isValid: boolean;
  errors: FieldErrors<SubscriptionForm>;
};
const SubscribeForm = ({
  state,
  control,
  isValid,
  errors,
}: SubscribeFormProps) => {
  const { pending } = useFormStatus();
  return (
    <>
      <Controller
        control={control}
        name="email"
        render={({ field: { onChange, onBlur, value } }) => (
          <>
            <Input
              name="email"
              placeholder="Your Email"
              classNames={{
                inputWrapper:
                  "max-w-[80%] sm:max-w-172 self-center bg-foreground-200 py-3.75 px-4 !rounded-1.25 border border-foreground-300 h-auto relative",
                input: "text-foreground-100 text-sm/7 h-auto",
                errorMessage: "text-center text-md",
              }}
              endContent={
                <Button
                  size="md"
                  type="submit"
                  className="hidden sm:block px-5.625 text-white text-sm/7 absolute right-0 rounded-l-none rounded-r-1.25"
                  isDisabled={!isValid || pending}
                  data-testid="subscribe-button"
                >
                  Subscribe
                </Button>
              }
              value={value}
              onChange={onChange}
              onBlur={onBlur}
              isInvalid={!!errors?.email || !!state?.errors?.email.length}
              errorMessage={errors?.email?.message}
              isDisabled={pending}
            />

            {state?.errors?.email &&
              state?.errors?.email?.length > 0 &&
              state.errors.email.map((error, index) => (
                <p key={index} className="text-md text-danger text-center">
                  {error}
                </p>
              ))}
          </>
        )}
      />

      <Button
        size="md"
        type="submit"
        className="sm:hidden px-5.625 text-white text-sm/7 w-full max-w-[80%]"
        isDisabled={!isValid || pending}
      >
        Subscribe
      </Button>
    </>
  );
};

const SubscribeSection = () => {
  const initialState: FormState<Subscription> = {};

  const [state, formAction] = useFormState<FormState<Subscription>, FormData>(
    subscribe,
    initialState,
  );

  const {
    control,
    formState: { isValid, errors },
    reset,
  } = useForm<SubscriptionForm>({
    mode: "onChange",
    defaultValues: {
      email: "",
    },
    resolver: zodResolver(SubscribeSchema),
  });

  useEffect(() => {
    if (state?.message && state?.status === FORM_STATUS.SUCCESS) {
      toast.success(state?.message);

      reset();
    }
  }, [reset, state.message, state?.status]);

  return (
    <section>
      <div className="max-w-xs sm:max-w-lg md:max-w-xl lg:max-w-4xl xl:max-w-5xl mx-auto my-0 py-40 flex flex-col gap-20 justify-center">
        <div className="flex flex-col gap-2.5 text-center">
          <p className="text-primary text-sm/6 font-bold">Newsletter</p>
          <h3 className="text-foreground text-2xl font-bold">
            Most Popular Courses
          </h3>
          <p className="text-foreground-100 text-sm whitespace-pre-line">
            {`Problems trying to resolve the conflict between
            the two major realms of Classical physics: Newtonian mechanics`}
          </p>
        </div>
        <form action={formAction} className="flex flex-col gap-2 items-center">
          <SubscribeForm
            control={control}
            errors={errors}
            isValid={isValid}
            state={state}
          />
        </form>
      </div>
    </section>
  );
};

export default SubscribeSection;
