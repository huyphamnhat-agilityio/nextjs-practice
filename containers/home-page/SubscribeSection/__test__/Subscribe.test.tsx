import { render, screen } from "@testing-library/react";
import { useFormState, useFormStatus } from "react-dom";

// Components
import SubscribeSection from "..";

// Constants
import { FORM_STATUS, SUBSCRIBE_MESSAGES } from "@/constants";

jest.mock("react-dom", () => ({
  ...jest.requireActual("react-dom"),
  useFormStatus: jest.fn(),
  useFormState: jest.fn(),
}));

describe("SubscribeSection test cases", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  const mockUseFormStatus = useFormStatus as jest.Mock;
  const mockUseFormState = useFormState as jest.Mock;

  mockUseFormState.mockReturnValueOnce([{}, () => {}]);

  const setup = () => render(<SubscribeSection />);

  it("should render correctly", () => {
    mockUseFormState.mockReturnValue([{}, () => {}]);

    mockUseFormStatus.mockReturnValue({
      pending: false,
    });

    const { asFragment } = setup();

    expect(asFragment()).toMatchSnapshot();
  });

  it("should render correctly when the form is pending", () => {
    mockUseFormState.mockReturnValue([{}, () => {}]);

    mockUseFormStatus.mockReturnValue({
      pending: true,
    });

    setup();

    const emailInput = screen.getByRole("textbox", {
      name: /your email/i,
    });

    const submitButton = screen.getByTestId("subscribe-button");

    expect(emailInput).toBeDisabled();
    expect(submitButton).toBeDisabled();
  });

  it("should show success toast when subscribing is successful", () => {
    mockUseFormState.mockReturnValue([
      {
        message: SUBSCRIBE_MESSAGES.SUCCESS,
        status: FORM_STATUS.SUCCESS,
        resetKey: Date.now().toString(),
      },
      () => {},
    ]);

    mockUseFormStatus.mockReturnValue({
      pending: false,
    });

    const { asFragment } = setup();

    expect(asFragment()).toMatchSnapshot();
  });

  it("should show error message when subscribing email is invalid", () => {
    mockUseFormState.mockReturnValue([
      {
        errors: {
          email: [SUBSCRIBE_MESSAGES.ERROR.REQUIRED],
        },
      },
      () => {},
    ]);

    mockUseFormStatus.mockReturnValue({
      pending: false,
    });

    setup();

    const errorMessage = screen.getByText(SUBSCRIBE_MESSAGES.ERROR.REQUIRED);

    expect(errorMessage).toBeInTheDocument();
  });
});
