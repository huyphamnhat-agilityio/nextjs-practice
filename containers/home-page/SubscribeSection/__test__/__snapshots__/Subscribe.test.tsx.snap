// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubscribeSection test cases should render correctly 1`] = `
<DocumentFragment>
  <section>
    <div
      class="max-w-xs sm:max-w-lg md:max-w-xl lg:max-w-4xl xl:max-w-5xl mx-auto my-0 py-40 flex flex-col gap-20 justify-center"
    >
      <div
        class="flex flex-col gap-2.5 text-center"
      >
        <p
          class="text-primary text-sm/6 font-bold"
        >
          Newsletter
        </p>
        <h3
          class="text-foreground text-2xl font-bold"
        >
          Most Popular Courses
        </h3>
        <p
          class="text-foreground-100 text-sm whitespace-pre-line"
        >
          Problems trying to resolve the conflict between
            the two major realms of Classical physics: Newtonian mechanics
        </p>
      </div>
      <form
        class="flex flex-col gap-2 items-center"
      >
        <div
          class="group flex flex-col data-[hidden=true]:hidden w-full group relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_10px)]"
          data-filled="true"
          data-filled-within="true"
          data-slot="base"
        >
          <div
            class="h-full flex flex-col"
            data-slot="main-wrapper"
          >
            <div
              class="w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm gap-3 data-[hover=true]:bg-default-200 group-data-[focus=true]:bg-default-100 min-h-10 rounded-small transition-background motion-reduce:transition-none !duration-150 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background rounded-1.25 max-w-[80%] sm:max-w-172 self-center bg-foreground-200 py-3.75 px-4 !rounded-1.25 border border-foreground-300 h-auto relative"
              data-slot="input-wrapper"
              style="cursor: text;"
            >
              <div
                class="inline-flex w-full items-center h-full box-border"
                data-slot="inner-wrapper"
              >
                <input
                  aria-label="Your Email"
                  class="w-full font-normal bg-transparent !outline-none placeholder:text-foreground-500 focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text group-data-[has-value=true]:text-default-foreground text-foreground-100 text-sm/7 h-auto"
                  data-has-end-content="true"
                  data-slot="input"
                  id="react-aria-:r0:"
                  name="email"
                  placeholder="Your Email"
                  type="text"
                  value=""
                />
                <button
                  class="z-0 group items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 min-w-20 gap-2 rounded-medium opacity-disabled pointer-events-none [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-primary data-[hover=true]:opacity-hover py-3.75 w-auto h-auto rounded-1.25 hidden sm:block px-5.625 text-white text-sm/7 absolute right-0 rounded-l-none rounded-r-1.25"
                  data-disabled="true"
                  data-testid="subscribe-button"
                  disabled=""
                  type="submit"
                >
                  Subscribe
                </button>
              </div>
            </div>
          </div>
        </div>
        <button
          class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 min-w-20 gap-2 rounded-medium opacity-disabled pointer-events-none [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-primary data-[hover=true]:opacity-hover py-3.75 h-auto rounded-1.25 sm:hidden px-5.625 text-white text-sm/7 w-full max-w-[80%]"
          data-disabled="true"
          disabled=""
          type="submit"
        >
          Subscribe
        </button>
      </form>
    </div>
  </section>
</DocumentFragment>
`;

exports[`SubscribeSection test cases should show error toast when subscribing fails 1`] = `
<DocumentFragment>
  <section>
    <div
      class="max-w-xs sm:max-w-lg md:max-w-xl lg:max-w-4xl xl:max-w-5xl mx-auto my-0 py-40 flex flex-col gap-20 justify-center"
    >
      <div
        class="flex flex-col gap-2.5 text-center"
      >
        <p
          class="text-primary text-sm/6 font-bold"
        >
          Newsletter
        </p>
        <h3
          class="text-foreground text-2xl font-bold"
        >
          Most Popular Courses
        </h3>
        <p
          class="text-foreground-100 text-sm whitespace-pre-line"
        >
          Problems trying to resolve the conflict between
            the two major realms of Classical physics: Newtonian mechanics
        </p>
      </div>
      <form
        class="flex flex-col gap-2 items-center"
      >
        <div
          class="group flex flex-col data-[hidden=true]:hidden w-full group relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_10px)]"
          data-filled="true"
          data-filled-within="true"
          data-slot="base"
        >
          <div
            class="h-full flex flex-col"
            data-slot="main-wrapper"
          >
            <div
              class="w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm gap-3 data-[hover=true]:bg-default-200 group-data-[focus=true]:bg-default-100 min-h-10 rounded-small transition-background motion-reduce:transition-none !duration-150 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background rounded-1.25 max-w-[80%] sm:max-w-172 self-center bg-foreground-200 py-3.75 px-4 !rounded-1.25 border border-foreground-300 h-auto relative"
              data-slot="input-wrapper"
              style="cursor: text;"
            >
              <div
                class="inline-flex w-full items-center h-full box-border"
                data-slot="inner-wrapper"
              >
                <input
                  aria-label="Your Email"
                  class="w-full font-normal bg-transparent !outline-none placeholder:text-foreground-500 focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text group-data-[has-value=true]:text-default-foreground text-foreground-100 text-sm/7 h-auto"
                  data-has-end-content="true"
                  data-slot="input"
                  id="react-aria-:rf:"
                  name="email"
                  placeholder="Your Email"
                  type="text"
                  value=""
                />
                <button
                  class="z-0 group items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 min-w-20 gap-2 rounded-medium opacity-disabled pointer-events-none [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-primary data-[hover=true]:opacity-hover py-3.75 w-auto h-auto rounded-1.25 hidden sm:block px-5.625 text-white text-sm/7 absolute right-0 rounded-l-none rounded-r-1.25"
                  data-disabled="true"
                  data-testid="subscribe-button"
                  disabled=""
                  type="submit"
                >
                  Subscribe
                </button>
              </div>
            </div>
          </div>
        </div>
        <button
          class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 min-w-20 gap-2 rounded-medium opacity-disabled pointer-events-none [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-primary data-[hover=true]:opacity-hover py-3.75 h-auto rounded-1.25 sm:hidden px-5.625 text-white text-sm/7 w-full max-w-[80%]"
          data-disabled="true"
          disabled=""
          type="submit"
        >
          Subscribe
        </button>
      </form>
    </div>
  </section>
</DocumentFragment>
`;

exports[`SubscribeSection test cases should show success toast when subscribing is successful 1`] = `
<DocumentFragment>
  <section>
    <div
      class="max-w-xs sm:max-w-lg md:max-w-xl lg:max-w-4xl xl:max-w-5xl mx-auto my-0 py-40 flex flex-col gap-20 justify-center"
    >
      <div
        class="flex flex-col gap-2.5 text-center"
      >
        <p
          class="text-primary text-sm/6 font-bold"
        >
          Newsletter
        </p>
        <h3
          class="text-foreground text-2xl font-bold"
        >
          Most Popular Courses
        </h3>
        <p
          class="text-foreground-100 text-sm whitespace-pre-line"
        >
          Problems trying to resolve the conflict between
            the two major realms of Classical physics: Newtonian mechanics
        </p>
      </div>
      <form
        class="flex flex-col gap-2 items-center"
      >
        <div
          class="group flex flex-col data-[hidden=true]:hidden w-full group relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_10px)]"
          data-filled="true"
          data-filled-within="true"
          data-slot="base"
        >
          <div
            class="h-full flex flex-col"
            data-slot="main-wrapper"
          >
            <div
              class="w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm gap-3 data-[hover=true]:bg-default-200 group-data-[focus=true]:bg-default-100 min-h-10 rounded-small transition-background motion-reduce:transition-none !duration-150 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background rounded-1.25 max-w-[80%] sm:max-w-172 self-center bg-foreground-200 py-3.75 px-4 !rounded-1.25 border border-foreground-300 h-auto relative"
              data-slot="input-wrapper"
              style="cursor: text;"
            >
              <div
                class="inline-flex w-full items-center h-full box-border"
                data-slot="inner-wrapper"
              >
                <input
                  aria-label="Your Email"
                  class="w-full font-normal bg-transparent !outline-none placeholder:text-foreground-500 focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text group-data-[has-value=true]:text-default-foreground text-foreground-100 text-sm/7 h-auto"
                  data-has-end-content="true"
                  data-slot="input"
                  id="react-aria-:ra:"
                  name="email"
                  placeholder="Your Email"
                  type="text"
                  value=""
                />
                <button
                  class="z-0 group items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 min-w-20 gap-2 rounded-medium opacity-disabled pointer-events-none [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-primary data-[hover=true]:opacity-hover py-3.75 w-auto h-auto rounded-1.25 hidden sm:block px-5.625 text-white text-sm/7 absolute right-0 rounded-l-none rounded-r-1.25"
                  data-disabled="true"
                  data-testid="subscribe-button"
                  disabled=""
                  type="submit"
                >
                  Subscribe
                </button>
              </div>
            </div>
          </div>
        </div>
        <button
          class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 min-w-20 gap-2 rounded-medium opacity-disabled pointer-events-none [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-primary data-[hover=true]:opacity-hover py-3.75 h-auto rounded-1.25 sm:hidden px-5.625 text-white text-sm/7 w-full max-w-[80%]"
          data-disabled="true"
          disabled=""
          type="submit"
        >
          Subscribe
        </button>
      </form>
    </div>
  </section>
</DocumentFragment>
`;
